docs/tutorials/sparse/train_gluon.md

```diff
@@ -0,0 +1,469 @@
+
+
+```python
+net = mx.gluon.nn.Sequential()
+net.add(
+    FullyConnectedSparse(in_units=num_features, units=128, weight_grad_stype='row_sparse'),
+    mx.gluon.nn.Activation('sigmoid'),
+    FullyConnected(in_units=128, units=8),
+    mx.gluon.nn.Activation('sigmoid'),
+    FullyConnected(in_units=8, units=1),
+    mx.gluon.nn.Activation('sigmoid'),
+)
+net.initialize(ctx=ctx)
+trainer = mx.gluon.Trainer(net.collect_params(), 'sgd')
+loss_fn = mx.gluon.loss.SigmoidBinaryCrossEntropyLoss()
+```
+
+
+```python
+%%timeit
+data_iter.reset()
+for batch in data_iter:
+    data = batch.data[0]
+    data = data.as_in_context(ctx)
+    label = batch.label[0].as_in_context(ctx)
+    with mx.autograd.record():
+        pred = net(data)
+        loss = loss_fn(pred, label)
+    loss.backward()
+    trainer.step(data.shape[0])
+    loss.wait_to_read()
+```
+
+    334 ms ± 16.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
+
+
+![fully_connected_sparse_grad_backward](https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/doc/tutorials/ndarray/sparse/fully_connected_sparse_grad_backward.png)
+
+We can see a huge reduction in the time taken for the backward pass and parameter update step: 3.99ms vs 0.18ms. And this reduces the overall time of the epoch significantly. Our gradient consumes a much smaller amount of memory and means only a subset of parameters need updating as part of the `sgd_update` step. Some optimizers don't support sparse gradients however, so reference the specific optimizer's documentation for more details.
+
+
+```python
+print_memory_allocation(net, block_idxs=[0, 2, 4])
+```
+
+    Memory Allocation for Weight:
+    512.000 MBs ( 99.999%) for fullyconnectedsparse1                   
+      0.004 MBs (  0.001%) for fullyconnected2                         
+      0.000 MBs (  0.000%) for fullyconnected3                         
+    Memory Allocation for Weight Gradient:
+      0.059 MBs ( 93.490%) for fullyconnectedsparse1                   
+      0.004 MBs (  6.460%) for fullyconnected2                         
+      0.000 MBs (  0.050%) for fullyconnected3                         
+
+
+### Advanced: Sparse `weight`
+
+You can optimize this example further by setting the weight's `stype` to `'row_sparse'`, but whether `'row_sparse'` weights make sense or not will depends on your specific task. 
+
+# Conclusion
+
+As part of this tutorial, we learnt how to write sparse data to disk in LibSVM format and load it back in sparse batches with the [`LibSVMIter`](https://mxnet.incubator.apache.org/api/python/io/io.html?highlight=libsvmiter). We learnt how to improve the performance of Gluon's [`nn.Dense`](https://mxnet.incubator.apache.org/versions/master/api/python/gluon/nn.html?highlight=dense#mxnet.gluon.nn.Dense) on sparse arrays using `mx.nd.sparse`. And lastly, we set `grad_stype` to `'row_sparse'` to reduce the size of the gradient and speed up the parameter update step.
```

Comment:
```suggestion
As part of this tutorial, we learned how to write sparse data to disk in LibSVM format and load it back in sparse batches with the [`LibSVMIter`](https://mxnet.incubator.apache.org/api/python/io/io.html?highlight=libsvmiter). We learned how to improve the performance of Gluon's [`nn.Dense`](https://mxnet.incubator.apache.org/versions/master/api/python/gluon/nn.html?highlight=dense#mxnet.gluon.nn.Dense) on sparse arrays using `mx.nd.sparse`. And lastly, we set `grad_stype` to `'row_sparse'` to reduce the size of the gradient and speed up the parameter update step.
```
I can handle the s instead of z and the other mother-tongue 🇬🇧 idiosyncrasies, but `learnt`? Naw, dog. `Learned` just sounds so much better. 🇺🇸

Reply:
haha, i'll change, but only because you just won the *football*.